javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+
+      override def addAll(index: Int, c: Collection[_ <: E]): Boolean =
+        throw new UnsupportedOperationException
+
+      override def set(index: Int, element: E): E =
+        throw new UnsupportedOperationException
+
+      override def add(index: Int, element: E): Unit =
+        throw new UnsupportedOperationException
+
+      override def remove(index: Int): E =
+        throw new UnsupportedOperationException
+
+      override def listIterator(): ListIterator[E] =
+        new ImmutableListIterator(inner.listIterator())
+
+      override def listIterator(index: Int): ListIterator[E] =
+        new ImmutableListIterator(inner.listIterator(index))
+
+      override def subList(fromIndex: Int, toIndex: Int): List[E] =
+        unmodifiableList(super.subList(fromIndex, toIndex))
+    }
+
+    private[Collections] trait ImmutableMap[K, V, M <: Map[K, V]] extends WrappedMap[K, V, M] {
+
+      override def put(key: K, value: V): V =
+        throw new UnsupportedOperationException
+
+      override def remove(key: scala.Any): V =
+        throw new UnsupportedOperationException
+
+      override def putAll(m: Map[_ <: K, _ <: V]): Unit =
+        throw new UnsupportedOperationException
+
+      override def clear(): Unit =
+        throw new UnsupportedOperationException
+
+      override def keySet(): Set[K] =
+        unmodifiableSet(super.keySet)
+
+      override def values(): Collection[V] =
+        unmodifiableCollection(super.values)
+
+      override def entrySet(): Set[Map.Entry[K, V]] =
+        unmodifiableSet(super.entrySet)
+    }
+
+    private[Collections] trait ImmutableSortedMap[K, V] extends ImmutableMap[K, V, SortedMap[K, V]]
+                                                           with WrappedSortedMap[K, V] {
+      override def subMap(fromKey: K, toKey: K): SortedMap[K, V] =
+        unmodifiableSortedMap(super.subMap(fromKey, toKey))
+
+      override def headMap(toKey: K): SortedMap[K, V] =
+        unmodifiableSortedMap(super.headMap(toKey))
+
+      override def tailMap(fromKey: K): SortedMap[K, V] =
+        unmodifiableSortedMap(super.tailMap(fromKey))
+    }
+
+    private class ImmutableIterator[E, Iter <: Iterator[E]](protected val inner: Iter)
+        extends WrappedIterator[E, Iter] {
+      override def remove(): Unit = throw new UnsupportedOperationException
+    }
+
+    private class ImmutableListIterator[E](innerIterator: ListIterator[E])
+        extends ImmutableIterator[E, ListIterator[E]](innerIterator)
+           with WrappedListIterator[E]{
+      override def set(e: E): Unit = throw new UnsupportedOperationException
+
+      override def add(e: E): Unit = throw new UnsupportedOperationException
+    }
+  }
+
+  private[Collections] object CheckedCollections {
+
+    private final def checkClass[T](elem: T, clazz: Class[T]): Unit = {
+      if (elem != null && !clazz.isInstance(elem))
+        throw new ClassCastException
```

Comment:
There is `clazz.cast(elem)` for that.

It is actually important to use `clazz.cast` and not `clazz.isInstance`, because the latter will not have the correct behavior for raw JS types.